manual/manual/cmds/intf-c.etex

```diff
@@ -2176,15 +2176,19 @@ CAMLprim stub_gethostbyname(value vname)
   CAMLparam1 (vname);
   CAMLlocal1 (vres);
   struct hostent * h;
+  char * name;
 
   /* Copy the string argument to a C string, allocated outside the
      OCaml heap. */
-  name = stat_alloc(caml_string_length(vname) + 1);
-  name = caml_strdup (String_val(vname));
+  name = caml_stat_alloc(caml_string_length(vname) + 1);
+  name = caml_strdup(String_val(vname));
   /* Release the OCaml run-time system */
   caml_release_runtime_system();
   /* Resolve the name */
   h = gethostbyname(name);
+  /* Free the copy of the string, which we might as well do before
+     acquiring the runtime system to benefit from parallelism. */
```

Comment:
I meant "freeing before acquiring the runtime system" as opposed to freeing after acquiring the runtime system, which is probably slightly more natural to write as it's more symmetric:

alloc(); release_runtime(); blocking_stuff(); acquire_runtime(); free()
vs
alloc(); release_runtime(); blocking_stuff(); free(); acquire_runtime()

But the more symmetric version doesn't allow ocaml code to run while free() is running.